### PR TITLE
Backports 0.18 pr1

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

File: src/wallet/coinselection.cpp [hunk 1]
```diff
@@ -268,7 +279,7 @@ bool KnapsackSolver(const CAmount& nTargetValue, std::vector<OutputGroup>& group
 
     if (fFulyMixedOnly) {
         // larger denoms first
-        std::sort(groups.rbegin(), groups.rend(), CompareByPriority());
+        std::sort(groups.rbegin(), groups.rend(), CompareByValue());
```
Comment (UdjinM6) [inline src/wallet/coinselection.cpp:282]: ?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
see below + should update circular dependencies with `"wallet/coincontrol -> wallet/wallet -> wallet/coincontrol"`

File: test/functional/test_runner.py [hunk 2]
```diff
@@ -130,6 +136,15 @@
...
     'p2p_invalid_tx.py',
+    'feature_assumevalid.py',
+    'example_test.py',
+    'wallet_txn_doublespend.py',
+    'wallet_txn_clone.py --mineblock',
+    'feature_notifications.py',
+    'rpc_invalidateblock.py',
+    'feature_rbf.py',
+    'mempool_packages.py',
+    'rpc_createmultisig.py',
```
Comment (UdjinM6) [inline test/functional/test_runner.py:147]:
```suggestion
```

File: src/wallet/coinselection.cpp (hunk 1 above)
Comment (Munkybooty) [reply src/wallet/coinselection.cpp:282]: there is no .Priority() with groups (formerly vCoins). #12257

File: test/functional/test_runner.py [hunk 3]
```diff
@@ -184,26 +198,8 @@
 EXTENDED_SCRIPTS = [
     # These tests are not run by the travis build process.
     # Longest test should go first, to favor running tests in parallel
-    'feature_pruning.py', # NOTE: Prune mode is incompatible with -txindex, should work with governance validation disabled though.
-    # vv Tests less than 20m vv
-    'feature_fee_estimation.py',
-    # vv Tests less than 5m vv
-    'feature_maxuploadtarget.py',
-    'mempool_packages.py',
+    'feature_pruning.py',
```
[the remark this answers is not in the document]
Comment (Munkybooty) [reply test/functional/test_runner.py:201]: Would you prefer this be commented into the code?

File: src/wallet/coinselection.cpp [hunk 4]
```diff
@@ -373,3 +360,40 @@ bool KnapsackSolver(const CAmount& nTargetValue, std::vector<CInputCoin>& vCoins
...
+    while (it != m_outputs.end() && it->outpoint != output.outpoint) ++it;
+    if (it == m_outputs.end()) return it;
+    m_value -= output.effective_value;
+    effective_value -= output.effective_value;
+    return m_outputs.erase(it);
+}
+
+bool OutputGroup::EligibleForSpending(const CoinEligibilityFilter& eligibility_filter) const
+{
+    return m_depth >= (m_from_me ? eligibility_filter.conf_mine : eligibility_filter.conf_theirs)
```
[the remark this answers is not in the document]
Comment (Munkybooty) [reply src/wallet/coinselection.cpp:407]: I can't seem to find logic for `CWallet::OutputEligibleForSpending`

**Review by PastaPastaPasta — DISMISSED** [review]
ACK, bitcoin#12257 is a little concerning, but I think everything is correct

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
I think 12257 needs a bit more work, pls see https://github.com/UdjinM6/dash/commits/pr4200

**Comment by UdjinM6** [comment]
I see that you squashed proposed changes into original PR (which is ok) but you left 7dad584 as an empty commit (which is not ok :) ). Pls drop 7dad584 and force-push the branch.

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
slightly tested ACK

**Pull request merged**